test/FunctionalTests/Framework/TestExecutionLayer/AstoriaDatabase.cs

```diff
@@ -173,7 +173,10 @@ public string DatabaseName
             {
                 if (_databaseName == null)
                 {
-                    _databaseName = String.Format("{0}_{1}_{2}", this.DatabasePrefixName, System.Net.Dns.GetHostName(), Guid.NewGuid().ToString("N"));
+                    _databaseName = String.Format("{0}_{1}_{2}",
+                        this.DatabasePrefixName,
+                        System.Net.Dns.GetHostName().Replace('-', '_'),

```

Comment:
A general theme should be converting other special characters as well as '-' here. Alternatively we can clarify the requirement for DNS hostname of not allowing special character. Here is the github issue submitted last week: https://github.com/OData/odata.net/issues/899